lib/logstash/outputs/elasticsearch.rb

```diff
@@ -176,7 +176,11 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # to service requests. If you have custom firewall rules you may need to change
   # this
   config :healthcheck_path, :validate => :string, :default => "/"
-
+  # When a `healthcheck_path` config is provided, this additional flag can be used to 
+  # specify whether it is an absolute URI or a relative path. 
+  # If this flag is true, `healthcheck_path` is expected to be a fully formed URL 
+  # with any basic auth credentials provided in the URL itself.
+  config :absolute_healthcheck_path, :validate => :boolean, :default => false
```

Comment:
in order for this setting to reach the pool class, it is necessary to pass it down through the http_client_builder and http_client too.